### PR TITLE
Add chroot option in multus-daemon

### LINKED ISSUE #829: '/var/run' should be '/run'.
**What would you like to be added**:

**Why is this needed**:

### PR DESCRIPTION
**Pull request by s1061123**
This change introduces 'chroot' option in multus-daemon to execute
delegate CNI with host filesystem context.

fix #829

### CODE REVIEW

File: deployments/multus-daemonset-thick.yml [hunk 1]
```diff
@@ -163,16 +164,19 @@ spec:
           volumeMounts:
             - name: cni
               mountPath: /host/etc/cni/net.d
-            - name: cnibin
```
Comment (dougbtv) [inline deployments/multus-daemonset-thick.yml:166]: Hey Tomo, why do we remove the bindir, but we keep the other directories? e.g. netns, confdir, etc.

File: deployments/multus-daemonset-thick.yml [hunk 2]
```diff
@@ -198,6 +202,9 @@ spec:
         - name: cnibin
           hostPath:
             path: /opt/cni/bin
+        - name: hostdir
```
Comment (dougbtv) [inline deployments/multus-daemonset-thick.yml:205]: maybe a more specific name, `hostrootdir` or something, seems kind of overloaded between `hostPath` and `hostdir`

File: deployments/multus-daemonset-thick.yml [hunk 3]
```diff
@@ -207,6 +214,9 @@ spec:
         - name: host-var-run
```
Comment (dougbtv) [inline deployments/multus-daemonset-thick.yml:214]: I realize this is unrelated to this changeset, but... I'm wondering if this /var/run mount is obsolete. I think we used to use it for a crio reboot hack? Maybe we can remove in 4.0

File: e2e/templates/multus-daemonset-thick.yml.j2 [hunk 4]
```diff
@@ -89,8 +89,8 @@ data:
         "logLevel": "debug",
         "logFile": "/tmp/multus.log",
         "binDir": "/host/opt/cni/bin",
```
Comment (dougbtv) [inline e2e/templates/multus-daemonset-thick.yml.j2:91]: should this still reference `/host/`?

File: deployments/multus-daemonset-thick.yml (hunk 2 above)
Comment (s1061123) [reply deployments/multus-daemonset-thick.yml:205]: Thanks. Changed to hostroot.

File: deployments/multus-daemonset-thick.yml (hunk 3 above)
Comment (s1061123) [reply deployments/multus-daemonset-thick.yml:214]: We still used to '/host/var/run/multus' for CNI cache. So we cannot remove it, but agreed that we could change to '/run'.
Let me add issue for that.

Comment (s1061123) [reply deployments/multus-daemonset-thick.yml:214]: Added as #829

File: deployments/multus-daemonset-thick.yml (hunk 1 above)
Comment (s1061123) [reply deployments/multus-daemonset-thick.yml:166]: In this yaml case (multus-thick with chroot), when the delegate CNI is executed, multus changed the root to host root, with chroot(), hence, we don't need to configure binDir for pod mount namespace. We just configure host directory in this case. Of course, we just use multus-daemon without chroot option, user need to configure.

File: deployments/multus-daemonset-thick.yml (hunk 3 above)
Comment (s1061123) [reply deployments/multus-daemonset-thick.yml]: It is taken care of the PR.

**Review by dougbtv — APPROVED** [review]
Thanks for the touch up on the directory as well as the explanations. LGTM!

**Pull request merged**